### ISSUE TITLE
Content and Software Directory

### ISSUE THREAD

**Issue by Miyitus** [miyovanstenis/the.wrong#25]
Please let keep it simple, 
the content folder should be inside of the software folder,

let's change the directory from Piratebox to The.wrong
too much folders can be confusing and we need the directory be only the.wrong 
we can keep piratebox inside on the conf files and so on but for us its important to keep the external part of the USB drive installation as easy as possible and with only the.wrong

**Comment by MaStr** [comment]
I totally understand your point.

TL;DR: If you want only the folder the.wrong and not piratebox on the USB stick, it will take at least 2 working days to achieve this.

The problem is, that the folder on the USB stick is created and used *before* the customization steps are executed. The Piratebox package comes with that configuration so, the steps are:

1. Install package content (including) /etc/piratebox.config
2. Do initialization via piratebox.config 
2.1 Create usb folders usb/PirateBox
2.2 Do Network configration
2.3. mount /opt/piratebox from img-file (located at usb/PirateBox)
2.4 Do run the shipped customization point  (this is the first point where I am able to change things in the running system)
2.5. Do OpenWrt customization
2.6 Do PirateBox initialization
2.7 Do first startup of PirateBox
2.8. Reboot.

The only way to get around this limitation is to create a new openwrt package called "thewrong", based on PirateBox with a custom configuration. Based on my current experience, it is not just copy, paste, replace, but typically there a bunch of follow up changes.

From the other point of view, you can say: Everything you need to know is inside the the.wrong folder, that's it. Technical stuff is inside PirateBox & the external_fs folder.


So, you can decide:
a. Leave it like it is, we have a very strong separation between content and technical stuff. We can still move over a few things.
b. Create the package and add delay. We are talking about 16 hours- (I'm working only a few hours per evening on this topic). If that change is faster made, lucky.. but I am calculating with the bad case scenario.

**Comment by Miyitus** [comment]
I understand how much time can get to change all path for the software, but I believe, from begining, that we were building « the.wrong »  based on Piratebox not a custom copy,

the « product » it has to be the.wrong because having two projects can be confusing, all the crédits are going to be there, we are going open about it but we need to keep it as The.wrong

I’m sorry if that wasn’t clear but I Believe you understood that from the beginning

If you need more time, it’s not a problem because this could be in the future something negatif for the project.

What I can do, I can help you to read all files and change the path, etc if you you provide me with the developer files you’re working on... let’s go option B I’m down to help you minimize work



 

📎 MVS 👁
http://miyovanstenis.com/

De : Matthias Strubel
Envoyé le :05 June 2019 21:49
À : miyovanstenis/the.wrong
Cc : Miyö Van Stenis; Author
Objet :Re: [miyovanstenis/the.wrong] Content and Software Directory (#25)

I totally understand your point.
TL;DR: If you want only the folder the.wrong and not piratebox on the USB stick, it will take at least 2 working days to achieve this.
The problem is, that the folder on the USB stick is created and used before the customization steps are executed. The Piratebox package comes with that configuration so, the steps are:
1. Install package content (including) /etc/piratebox.config
2. Do initialization via piratebox.config
2.1 Create usb folders usb/PirateBox
2.2 Do Network configration
2.3. mount /opt/piratebox from img-file (located at usb/PirateBox)
2.4 Do run the shipped customization point (this is the first point where I am able to change things in the running system)
2.5. Do OpenWrt customization
2.6 Do PirateBox initialization
2.7 Do first startup of PirateBox
2.8. Reboot.
The only way to get around this limitation is to create a new openwrt package called "thewrong", based on PirateBox with a custom configuration. Based on my current experience, it is not just copy, paste, replace, but typically there a bunch of follow up changes.
From the other point of view, you can say: Everything you need to know is inside the the.wrong folder, that's it. Technical stuff is inside PirateBox & the external_fs folder.
So, you can decide:
a. Leave it like it is, we have a very strong separation between content and technical stuff. We can still move over a few things.
b. Create the package and add delay. We are talking about 16 hours- (I'm working only a few hours per evening on this topic). If that change is faster made, lucky.. but I am calculating with the bad case scenario.

**Comment by MaStr** [comment]
The path is not hard coded, it is based on the configuration. The time consuming is the creation of another package, add it to the Makefile.

I understand your concerns. By today's morning, from my point I would have been able to solve everything with the the available customization points. But this directory change is hard.

I understood, that we want an easy way to provide the artist to add his files. For this, we are adding a user, which has its starting point directly in usb/the.wrong . 

If that is so much important, I'll take care of it.

**Comment by Miyitus** [comment]
Oh ! I was worry you have to change all path directories because I’ve done stuff like that with openwrt  and was a nightmare but yes if only re-create the package its more time  to the work anyway  

😊




📎 MVS 👁
http://miyovanstenis.com/

De : Matthias Strubel
Envoyé le :05 June 2019 22:29
À : miyovanstenis/the.wrong
Cc : Miyö Van Stenis; Author
Objet :Re: [miyovanstenis/the.wrong] Content and Software Directory (#25)

The path is not hard coded, it is based on the configuration. The time consuming is the creation of another package, add it to the Makefile.

**Comment by MaStr** [comment]
https://github.com/PirateBox-Dev/openwrt-dev-environment/commit/5b8b70d07e35a6c76e9c6e56b663486844d1ae39
https://github.com/PirateBox-Dev/openwrt-piratebox-feed/commit/d589b1da459d777026399dd4e952759dd2993abc
https://github.com/PirateBox-Dev/openwrt-image-build/commit/f6ed0f4a76ced2053e1c61d078b57472189b995c


Let's see what will happen on my next build test..

**Comment by MaStr** [comment]
Issues:

Config adjustment in thewrong package still does not work.
extendRoot-thewrong does not have correct package dependencies inside the package 
extendRoot-thewrong is not arch independend

**Comment by Miyitus** [comment]
cross fingersEnviado desde mi Huawei-------- Mensaje original --------Asunto: Re: [miyovanstenis/the.wrong] Content and Software Directory (#25)De: Matthias Strubel Para: "miyovanstenis/the.wrong" CC: Miyö Van Stenis ,Author PirateBox-Dev/openwrt-dev-environment@5b8b70d
PirateBox-Dev/openwrt-piratebox-feed@d589b1d
PirateBox-Dev/openwrt-image-build@f6ed0f4
Let's see what will happen on my next build test..

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/miyovanstenis/the.wrong/issues/25?email_source=notifications\u0026email_token=AAIWQ6YDVT35EJ3OCD6WHBLPZCVW3A5CNFSM4HTYD4EKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXB37OY#issuecomment-499367867",
"url": "https://github.com/miyovanstenis/the.wrong/issues/25?email_source=notifications\u0026email_token=AAIWQ6YDVT35EJ3OCD6WHBLPZCVW3A5CNFSM4HTYD4EKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXB37OY#issuecomment-499367867",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by MaStr** [comment]
Pkg build ok, tests are running

**Comment by MaStr** [comment]
Tests are ok. PirateBox folder completly vanished on the USB stick.
